lib/handlers/pools/pool-caching/v2/v2-dynamo-cache.ts

```diff
@@ -0,0 +1,100 @@
+import { ICache } from '@uniswap/smart-order-router/build/main/providers/cache'
+import { Pair } from '@uniswap/v2-sdk'
+import { DocumentClient } from 'aws-sdk/clients/dynamodb'
+import { log } from '@uniswap/smart-order-router'
+import { PairMarshaller } from '../../../marshalling'
+
+export class V2DynamoCache implements ICache<{ pair: Pair; block?: number }> {
+  private readonly ddbClient: DocumentClient
+  private readonly DEFAULT_TTL = 60 // 1 minute
+  constructor(private readonly tableName: string) {
+    this.ddbClient = new DocumentClient({
+      maxRetries: 1,
+      retryDelayOptions: {
+        base: 20,
+      },
+      httpOptions: {
+        timeout: 100,
+      },
+    })
+  }
+  async get(key: string): Promise<{ pair: Pair; block?: number } | undefined> {
+    try {
+      const queryParams = {
```

Comment:
Feels a bit inefficient to scan the index. Think we can directly use getParam without any scanning, since we either look up when the block number is there, or not.

Like the DynamoCachingV3Pool:

```
const getParams = {
        TableName: this.tableName,
        Key: {
          poolAddress: partitionKey,
          blockNumber: sortKey,
        },
      }

      const cachedPoolBinary: Buffer | undefined = (
        await this.ddbClient
          .get(getParams)
          .promise()
          .catch((error) => {
            log.error({ error, getParams }, `[DynamoCachingV3Pool] Cached pool failed to get`)
            return undefined
          })
      )?.Item?.item
```

Reply:
good call

Reply:
oh I remember now... we don't have the block here